### PR TITLE
SONAR-9334 make EVENTS.COMPONENT_UUID not nullable and clean orphans

### PR DESCRIPTION
**Pull request by sns-seb**
based on #2108

### CODE REVIEW

File: server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/PopulateEventsComponentUuid.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    long id = row.getLong(1);
+    String projectUuid = row.getString(2);
+
+    update.setString(1, projectUuid);
+    update.setLong(2, id);
+
+    return true;
+  }
+
+  private void deleteOrphans(Context context) throws SQLException {
```
Comment (SonarTech) [inline server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/PopulateEventsComponentUuid.java]: MAJOR Make "deleteOrphans" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

File: server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/PopulateEventsComponentUuid.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+  @Override
+  protected void execute(Context context) throws SQLException {
+    populateColumnOrFixInconsistencies(context);
+
+    deleteOrphans(context);
+
+  }
+
+  private void populateColumnOrFixInconsistencies(Context context) throws SQLException {
```
Comment (SonarTech) [inline server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/PopulateEventsComponentUuid.java]: MAJOR Make "populateColumnOrFixInconsistencies" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

File: server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/MakeEventsComponentUuidNotNullable.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  public MakeEventsComponentUuidNotNullable(Database db) {
+    super(db);
+  }
+
+  @Override
+  public void execute(Context context) throws SQLException {
+    context.execute(new AlterColumnsBuilder(getDialect(), "events")
+      .updateColumn(VarcharColumnDef.newVarcharColumnDefBuilder()
+        .setColumnName("component_uuid")
+        .setLimit(VarcharColumnDef.UUID_VARCHAR_SIZE)
```
Comment (SonarTech) [inline server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/MakeEventsComponentUuidNotNullable.java]: MINOR Remove this use of "UUID_VARCHAR_SIZE"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/MakeEventsComponentUuidNotNullable.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  @Override
+  public void execute(Context context) throws SQLException {
+    context.execute(new DropIndexBuilder(getDialect())
+      .setTable(TABLE_EVENTS)
+      .setName(INDEX_EVENTS_COMPONENT_UUID)
+      .build());
+
+    VarcharColumnDef componentUuidColumnDef = VarcharColumnDef.newVarcharColumnDefBuilder()
+      .setColumnName("component_uuid")
+      .setLimit(VarcharColumnDef.UUID_VARCHAR_SIZE)
```
Comment (SonarTech) [inline server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/MakeEventsComponentUuidNotNullable.java]: MINOR Remove this use of "UUID_VARCHAR_SIZE"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/MakeEventsComponentUuidNotNullable.java (hunk 4 above)
Comment (sns-seb) [inline server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/MakeEventsComponentUuidNotNullable.java]: Yes, I know, but when the column was defined, this was the property in use and it has the correct size (50). The replacing property has size 40. So, this is the property that should be used here.

File: server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/RecreateIndexEventsComponentUuid.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+  @Override
+  public void execute(Context context) throws SQLException {
+    context.execute(new CreateIndexBuilder(getDialect())
+      .setTable("events")
+      .setName("events_component_uuid")
+      .setUnique(false)
+      .addColumn(newVarcharColumnDefBuilder()
+        .setColumnName("component_uuid")
+        .setLimit(VarcharColumnDef.UUID_VARCHAR_SIZE)
```
Comment (SonarTech) [inline server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/RecreateIndexEventsComponentUuid.java]: MINOR Remove this use of "UUID_VARCHAR_SIZE"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request closed without merging**